contracts/HealthBlock.sol

```diff
@@ -28,6 +28,67 @@ contract HealthBlock {
         address id;
     }
 
+    event DoctorRequestRaised(address indexed doctor, string indexed doctorName, string credentialsHash);
+    event RequestApproved(address indexed doctor, address indexed provider, uint256 indexed requestId);
+    event RequestRejected(address indexed doctor, address indexed provider, uint256 indexed requestId);
+
+    struct Request {
+        address doctor;
+        string doctorName;
+        string credentialsHash;
+        bool approved;
+    }
+    mapping(address => bool) private isDoctorVerified;
```

Comment:
can we have this parameter of Doctor verified inside doctor details? with information regarding the entity who verified the doctor?